url/src/parser.rs

```diff
@@ -947,12 +957,13 @@ impl<'a> Parser<'a> {
         Ok((username_end, remaining))
     }
 
+    #[allow(clippy::type_complexity)]
```

Comment:
If use the `enum` from above I could potentially pass it down from here, reducing the complexity.